controllers/main.js

```diff
@@ -0,0 +1,120 @@
+const express = require("express");
+const router = express.Router();
+const pool = require("../db.js");
+
+router.get("/", async (req, res) => {
+  const title = "main";
+  if (req.session.user) {
+    const nickname = req.session.user["nickname"];
+    let user_id= await pool.query(`SELECT id FROM user WHERE nickname=? `, [nickname]);
+    user_id= user_id[0][0].id;
+    const goal_list = await pool.query(`SELECT * FROM goal`);
+    const cheer_list = await pool.query(`SELECT * FROM cheer`);
+
+    //응원한 목표들 리스트
+   let cheer_goal_list = await pool.query("SELECT g.goal_id, g.goal_content FROM user AS u LEFT OUTER JOIN cheer AS c ON u.id = c.cheer_user_id LEFT OUTER JOIN goal AS g ON c.cheer_goal_id = g.goal_id WHERE u.id = ?",[user_id]);
+   let arr = new Array();
+   for(let i=0; i<cheer_goal_list[0].length;i++){
+      arr[i]=cheer_goal_list[0][i].goal_id;
+   }
+   arr.sort(function(a, b)  {
+    return b - a;
+  });
+
+
+    res.render("main", {
+      title: title,
+      nickname: nickname,
+      user_id: user_id,
+      goal_list: goal_list[0],
+      cheer_list: cheer_list[0],
+      arr: arr
+
+    });
+  } else {
+    res.render("login", {
+      title: title,
+    });
+  }
```

Comment:
```suggestion
  try{
  if (req.session.user) {
    const nickname = req.session.user["nickname"];
    let user_id= await pool.query(`SELECT id FROM user WHERE nickname=? `, [nickname]);
    user_id= user_id[0][0].id;
    const goal_list = await pool.query(`SELECT * FROM goal`);
    const cheer_list = await pool.query(`SELECT * FROM cheer`);

    //응원한 목표들 리스트
   let cheer_goal_list = await pool.query("SELECT g.goal_id, g.goal_content FROM user AS u LEFT OUTER JOIN cheer AS c ON u.id = c.cheer_user_id LEFT OUTER JOIN goal AS g ON c.cheer_goal_id = g.goal_id WHERE u.id = ?",[user_id]);
   let arr = new Array();
   for(let i=0; i<cheer_goal_list[0].length;i++){
      arr[i]=cheer_goal_list[0][i].goal_id;
   }
   arr.sort(function(a, b)  {
    return b - a;
  });


    res.render("main", {
      title: title,
      nickname: nickname,
      user_id: user_id,
      goal_list: goal_list[0],
      cheer_list: cheer_list[0],
      arr: arr

    });
  } else {
    res.render("login", {
      title: title,
    });
  } } catch(err) {
     console.error(err)
  }
```
async await 문을 사용하기 때문에 try-catch문을 통해 예외처리를 해주는 것이 좋을 것 같습니다.

Reply:
확인했습니다.